### ISSUE TITLE
Absolute path for external .yml files

### ISSUE THREAD

**Issue by gcampionpae** [serverless/serverless#7864]
### Use case description

We have many repositories using very similar serverless.yml files for deployment and it would be nice if we could template some of the variables or sections of the serverless.yml so that when we have to change something for all serverless templates, we could do this in one place outside of the repository.  i.e.

```
plugins: ${file(https://gitlab.com/xxxxx/serverless-template-test/-/raw/master/serverless-plugins.yml)}
```

where the serverless-plugins.yml looks like this:
```
  localPath: /sls_plugins/node_modules
  modules:
    - serverless-python-requirements
    - serverless-plugin-git-variables
    - serverless-prune-plugin
```

**Comment by medikoo** [comment]
@gcampionpae I believe the problem is that variables resolution doesn't work for `plugins` sections.

That problem is handled at https://github.com/serverless/serverless/issues/7207

**Comment by gcampionpae** [comment]
Hi @medikoo, thanks for getting back to me so soon.  I think this also has to do with the PromiseTracker here: 
https://github.com/serverless/serverless/blob/4c6f8be5ccae88034e19f72a53996208dd4a56d5/lib/classes/PromiseTracker.js

Since it's not able to resolve the https file.. 

I tried to use an external file for the custom section as well and got this error as a result  
```
custom: ${file(https://gitlab.com/xxxxx/serverless-template-test/-/raw/master/serverless.yml)}
```

```
  Invalid variable reference syntax for variable file(https://gitlab.com/xxxxx/serverless-template-test/-/raw/master/serverless.yml). You can only reference env vars, options, & files. You can check our docs for more info.

```

**Comment by medikoo** [comment]
`file` is purely for File system resolution, it doesn't support url's.

At this point `url` resolution is not supported, but I believe we can consider such support (via e.g. `url:` prefix). Do you want to enquiry for such support?

**Comment by gcampionpae** [comment]
yeah that would be awesome, I wouldn't mind working on it as well if you can help point me in the right direction ;)

**Comment by medikoo** [comment]
@gcampionpae I'ev updated issue title, and added _Proposed solution_ section to main descroption, which provides general notes on how it can be implemented.

PR is welcome

**Comment by gcampionpae** [comment]
Hi @medikoo I think that until #7207 is resolved, this wouldn't help us with our idea for having a primary template for all of our projects.  Is there any timeline on when that will be resolved?

**Comment by medikoo** [comment]
>  Is there any timeline on when that will be resolved?

Unfortunately no. It's not considered a priority at this point
